docs/rules/no-type-literal.md

```diff
@@ -0,0 +1,121 @@
+# Enforces the use of interfaces over type literals (`type T = {...}`).
```

Comment:
correct me if I'm wrong, isn't type aliases a much broader concept? a type alias could be:
- an actual alias `type foo = string | string[];`
- a callback `type foo = () => 'hello';`
- a literal `type foo = {};`

I used type literals, because the rule only covers the last case...

anyway I'm ok changing the text, just want to make sure you are ok with this